bsdbpf/bsd_bpf_sniffer.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2012 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+// +build darwin dragonfly freebsd netbsd openbsd
+
+package bsdbpf
+
+import (
+	"github.com/google/gopacket"
+	"golang.org/x/sys/unix"
+
+	"fmt"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+const wordSize = int(unsafe.Sizeof(uintptr(0)))
+
+func bpfWordAlign(x int) int {
+	return (((x) + (wordSize - 1)) &^ (wordSize - 1))
+}
+
+type Options struct {
+	bpfDeviceName    string
```

Comment:
You'll need to export (capitalize) these names, so users of your API can set them.  Also, please add a comment to each detailing what they do